types/pg-cursor/index.d.ts

```diff
@@ -0,0 +1,31 @@
+// Type definitions for pg-cursor 2.5
+// Project: https://github.com/brianc/node-postgres#readme
+// Definitions by: Tiogshi Laj <https://github.com/Tiogshi>, Edited by Elias Fauser <https://github.com/elias-fauser>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+/// <reference types="node" />
+
+import { Connection, CustomTypesConfig, QueryResult } from "pg";
+
+declare namespace Cursor {
+    interface CursorQueryConfig {
+        /**
+         * By default rows come out as a Record<string,any>.
+         * Pass the string 'array' here to receive each row as an array of values.
+         */
+        rowMode?: "array";
+        /**
+         * Custom type parsers for just this query result.
+         */
+        types?: CustomTypesConfig;
+    }
+    type ResultCallback<RowType> = (err: Error | undefined, rows: RowType[], result: QueryResult) => void;
+}
+
+declare class Cursor<Row = any> {
```

Comment:
The class should be extended from EventEmitter. Please see the source code
https://github.com/brianc/node-postgres/blob/master/packages/pg-cursor/index.js#L9

Reply:
You should define the deprecated function as well.
https://github.com/brianc/node-postgres/blob/master/packages/pg-cursor/index.js#L254